### DISCUSSION TITLE
Select an emote from the emote window with only the keyboard

### DISCUSSION THREAD

**Discussion by GForceWeb** [Chatterino/chatterino2#3928] [Ideas]
Apologies if this is covered somewhere that I've missed but I'd really like to be able to open the emote window, type my search and then navigate through the results and add my chosen emotes to my message using only the keyboard and it doesn't seem to be possible.

I can open the emote window with Ctrl+E, then type my search to find what I'm looking for, but then there doesn't seem to be any way to add an emote from this window to my message without using the mouse.

Would be ideal if I could use Tab to jump into the results, then use arrow keys to navigate the results and add the selected emote to my message with Enter

Sidenote: Is it possible to have emotes render in the message compose box the same way native twitch chat does?

**Comment by pajlada** [comment]
I'm ok with the idea of allowing tab-navigating to emotes on the emote menu, and then if an emote is selected Enter should add the emote as if you clicked it.

Pressing Enter should not close the emote menu, so you would need to press Escape when you're done adding emotes. 

Tab-navigating is relatively simple to reason about in the search view<sup>a</sup>, so I think there it makes sense to allow. Should tab-navigating also be enabled in the default view<sup>b</sup>?
If tab-navigating is enabled in the default view, should the emote tabs be ignored when tab-navigating?


I think arrow keys should not be used for this, rather we should use Tab and Shift-Tab to navigate forward and backwards among all possible emotes.
Up Arrow and Down Arrow do not have an intuitive action as emotes are not rendered in a grid, but in a long list that may or may not wrap depending on the amount of emotes and , we render a list of emote and depending on the size of the window they may be rendered above or below each other.


<sup>a</sup>Search view: ![search view](https://user-images.githubusercontent.com/962989/185223664-ed7ebc1c-ec17-48d9-a697-5dd687882f4b.png)
<sup>b</sup>Default view: ![default view](https://user-images.githubusercontent.com/962989/185223608-7763fa14-85ed-4e6b-a957-0cc49a2ccb12.png)

**Reply by GForceWeb** [reply]
> Pressing Enter should not close the emote menu, so you would need to press Escape when you're done adding emotes.
> 
> Tab-navigating is relatively simple to reason about in the search viewa, so I think there it makes sense to allow. Should tab-navigating also be enabled in the default viewb?
> If tab-navigating is enabled in the default view, should the emote tabs be ignored when tab-navigating?
> 
> I think arrow keys should not be used for this, rather we should use Tab and Shift-Tab to navigate forward and backwards among all possible emotes.

Agree with all these points, Enter should only add the emote to your message, it shouldn't close the emote window. And tab for cycling through the search results makes more sense since they aren't in a standardised grid format.

Given there are already Ctrl+1-4 hotkeys for the tabs in the default view I think it's ok to skip these in the default view for tab-navigating. Though in general tab-navigating emotes without first using a search seems less useful.

**Comment by Mm2PL** [comment]
17 sie 2022 20:22:29 Grant ***@***.***>:

> Apologies if this is covered somewhere that I've missed but I'd really like to be able to open the emote window, type my search and then navigate through the results and add my chosen emotes to my message using only the keyboard and it doesn't seem to be possible.
> 
> I can open the emote window with Ctrl+E, then type my search to find what I'm looking for, but then there doesn't seem to be any way to add an emote from this window to my message without using the mouse.

> 
> Would be ideal if I could use Tab to jump into the results, then use arrow keys to navigate the results and add the selected emote to my message with Enter
You can use :-completion to search emotes with a preview directly in the chat input. It's not exactly what you want but it may be good enough.

> 
> Sidenote: Is it possible to have emotes render in the message compose box the same way native twitch chat does?
Nope. We do not have wysiwyg editing.

**Reply by GForceWeb** [reply]
> You can use :-completion to search emotes with a preview directly in the chat input. It's not exactly what you want but it may be good enough.

Right, I checked the option for emote autocomplete and couldn't actually work out how to trigger it because I didn't fully read/comprehend the "by typing :" bit at the end. I was trying to tab complete it and also spent time looking through the hotkey list for a way to trigger it. This does kinda fulfil what I was trying to do. There may still be times when having that tab-navigation in the full emote browser would be useful but this will certainly serve me well in the meantime. Thanks for pointing that out.
